nix/lib.nix

```diff
@@ -39,6 +39,7 @@ rec {
   evalModules = { name ? "process-compose", modules }: (lib.evalModules {
     specialArgs = {
       inherit name pkgs;
+      process-compose-flake-lib = (import ./process-compose-flake-lib.nix) { lib = pkgs.lib; };
```

Comment:
I'd expect a special arg like `process-compose-flake-lib` to point to `./nix/lib.nix`. Besides, as commented elsewhere, a function like `mkProcessComposeArgumentsOption` seems redundant (unless I'm missing something), and that you can simply import that submodule directly for re-use.